pkg/antctl/raw/check/cluster/test_checkcontrolplaneavailability.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type checkControlPlaneAvailability struct{}
+
+func init() {
+	RegisterTest("Check Control Plane Availability", &checkControlPlaneAvailability{})
+}
+
+func (t *checkControlPlaneAvailability) Run(ctx context.Context, testContext *testContext) error {
+	labelSelector := "component=kube-controller-manager,tier=control-plane"
+	namespaces, err := testContext.client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list Namespaces: %w", err)
+	}
+	controlPlaneFound := false
+	for _, namespace := range namespaces.Items {
+		pods, err := testContext.client.CoreV1().Pods(namespace.Name).List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+		if err != nil {
+			return fmt.Errorf("failed to list Pods in Namespace %s: %w", namespace.Name, err)
+		}
+		if len(pods.Items) > 0 {
+			controlPlaneFound = true
+			for _, pod := range pods.Items {
+				testContext.Log("Control plane Pod %s found in Namespace %s\n", pod.Name, namespace.Name)
+			}
+		}
+	}
+	if !controlPlaneFound {
+		testContext.Log("Warning: No control plane Pods found in any Namespace.")
```

Comment:
This is a bit complex but this message should really be more nuanced so as to avoid confusing users. The only thing we can really say here is:

> No control-plane Nodes were found; if installing Antrea in encap mode, some K8s functionalities (API aggregation, apiserver proxy, admission controllers) may be impacted.

Reply:
alright